exasol_transformers_extension/udfs/models/filling_mask_udf.py

```diff
@@ -136,65 +124,63 @@ def get_prediction(self, model_df: pd.DataFrame) -> pd.DataFrame:
         """
+        top_k = int(model_df['top_k'].iloc[0])
         text_data_raw = list(model_df['text_data'])
-        text_data_with_valid_mask_token = \
-            [text_data.replace(self.mask_token,
-                               self.last_created_pipeline.tokenizer.mask_token)
-             for text_data in text_data_raw]
-        results = self.last_created_pipeline(text_data_with_valid_mask_token)
+        text_data_with_valid_mask_token = [
+            text_data.replace(
+                self.mask_token,
+                self.last_created_pipeline.tokenizer.mask_token
+            ) for text_data in text_data_raw]
+        results = self.last_created_pipeline(
+            text_data_with_valid_mask_token, top_k=top_k)
 
         #  Batch prediction returns list of list while single prediction just
-        #  return a list. In case of batch predictions, we need to flatten
-        #  2D prediction results to 1D list
-        results = sum(results, []) if type(results[0]) == list else results
+        #  return a list. In order to ease dataframe operations, convert single
+        #  prediction to list of list.
+        results = [results] if len(text_data_raw) == 1 else results
 
-        filled_texts = []
-        scores = []
+        columns = ["sequence", "score"]
+        results_df_list = []
         for result in results:
-            filled_texts.append(result['sequence'])
-            scores.append(result['score'])
-
-        return scores, filled_texts
+            result_df = pd.DataFrame(result)
+            result_df = result_df[columns].rename(
+                columns={"sequence": "filled_text"})
+            results_df_list.append(result_df)
+        return results_df_list
 
     @staticmethod
     def _prepare_prediction_dataframe(
-            model_df: pd.DataFrame,
-            scores: List[float],
-            filled_texts: List[str]) -> pd.DataFrame:
+            model_df: pd.DataFrame, pred_df_list: List[pd.DataFrame]) \
+            -> pd.DataFrame:
         """
         Reformat the dataframe used in prediction, such that each input rows
         has a row for each label and its probability score
 
-        :param model_df: Dataframe used in prediction
-        :param scores: List of prediction probabilities
-        :param filled_texts: List of text filled with the predicted tokens
+       :param model_df: Dataframe used in prediction
+        :param pred_df_list: List of predictions dataframes
 
         :return: Prepared dataframe including input data and predictions
         """
-        top_k = model_df['top_k'].iloc[0]
-
-        # Repeat each row consecutively as the number of labels. At the end,
-        # the dataframe is expanded from (m, n) to (m*top_k, n)
-        repeated_indexes = model_df.index.repeat(top_k)
+        n_topk_results = list(map(lambda x: x.shape[0], pred_df_list))
+        repeated_indexes = model_df.index.repeat(repeats=n_topk_results)
```

Comment:
N_topk_results looks like a list of tuples of count, value. Does repeat handle this.

Reply:
yes, exactly